app/Services/Auth/ForgetPasswordService.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Services\Auth;
+
+use \App\Http\Requests\Auth\ForgetPasswordRequest;
+use App\Http\Resources\ErrorResource;
+use App\Http\Resources\SuccessResource;
+use App\Repositories\UserRepository;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Password;
+
+class ForgetPasswordService
+{
+    /**
+     * @var UserRepository
+     */
+    private $userRepo = NULL;
+
+    public function __construct(UserRepository $userRepo)
+    {
+        $this->userRepo = $userRepo;
+    }
+
+    /**
+     * @param ForgetPasswordRequest $request
+     * @return \Illuminate\Http\Resources\Json\JsonResource
+     */
+    public function execute(ForgetPasswordRequest $request)
+    {
+        try {
+            $user = $this->userRepo->findWhere(['email' => $request->get('email')])->first();
+            return $this->handleForgetPasswordEmail($request, $user);
+        }catch(\Exception $ex){
+            Log::info($ex);
+            return new ErrorResource(Response::HTTP_INTERNAL_SERVER_ERROR,
+                __('Failed to send reset password email!'));
+        }
+    }
+
+    private function handleForgetPasswordEmail(ForgetPasswordRequest $request, $user){
+        if ($user) {
+            $response = Password::sendResetLink(['email' => $request->get('email')]);
+            switch ($response) {
+                case Password::RESET_LINK_SENT:
+                    return new SuccessResource(Response::HTTP_OK, __('passwords.sent'));
+
+                case Password::INVALID_USER:
+                    return new ErrorResource(Response::HTTP_UNAUTHORIZED,
```

Comment:
I think this should not be unauthorized error, it should be unprocessable entity with 422 status code.

Reply:
@AbdullahMuhammed5 I think 401 is correct as there is something that went wrong with user credentials. Check this description from stack overflow
`If the request included authentication credentials, then the 401 response indicates that authorization has been refused for those credentials. The user agent MAY repeat the request with a new or replaced Authorization header field (Section 4.2).`
What do you think?

Reply:
The authentication credentials is email, phone and password or any credentials that request authorization on application data.
In my opinion forget password doesn't request any kind of authorizations, I just want to reset my password so i send you my email if it was wrong you send me that it's wrong.